### PR TITLE
Examples: Sound test: MegaDuck support, init wave ram

### PR DESCRIPTION
**Pull request by bbbbbr**

### CODE REVIEW

**Comment by untoxa** [comment]
hmmm... `nybble_swap_if_megaduck()`? suggest more generic names for that, like `platform_translate()`.

**Comment by bbbbbr** [comment]
Can change the naming if you'd prefer.

The reasoning for the names was:
- That program is very specific to the Game Boy sound hardware. Unlikely it is ported to a non-GB like console.
- The MegaDuck is the only known GB clone with register variations
- Choose function names which emphasizes what those differences are for easy understanding.
  - `nybble_swap_if_megaduck()`
  - `NR32_swap_volume_if_megaduck()`

**Comment by untoxa** [comment]
translate_volume(), translate_envelope(). why you put megaduck there? what if we need more than 1 ifdef there in future?

**Comment by bbbbbr** [comment]
I can rename them to be more general purpose, but the argument for it is not persuasive.

> why you put megaduck there? what if we need more than 1 ifdef there in future?

There is only one single known GB clone with register variations that exists: MegaDuck. What other purpose is anyone going to use an ifdef in there for?

**Comment by untoxa** [comment]
there could be two with analogue pocket though :)

**Comment by bbbbbr** [comment]
Let me rephrase that more specific - Analogue Pocket does not have Audio Hardware register address and bit flag changes.What would the ifdef for the sound hardware do?
